### ISSUE TITLE
CultureNotFoundException when decompiling a satellite assembly

### ISSUE THREAD

**Issue by pvoosten** [icsharpcode/ILSpy#2691]
Since version 7.2.0.6791-preview3, trying to extract the resources from a satellite assembly often results in the following stack trace:

```
System.Globalization.CultureNotFoundException: Only the invariant culture is supported in globalization-invariant mode. See https://aka.ms/GlobalizationInvariantMode for more information. (Parameter 'name')
it is an invalid culture identifier.
   at System.Globalization.CultureInfo..ctor(String name, Boolean useUserOverride)
   at System.Reflection.AssemblyName.set_CultureName(String value)
   at System.Reflection.Metadata.MetadataReader.GetAssemblyName(StringHandle nameHandle, Version version, StringHandle cultureHandle, BlobHandle publicKeyOrTokenHandle, AssemblyHashAlgorithm assemblyHashAlgorithm, AssemblyFlags flags)
   at System.Reflection.Metadata.AssemblyDefinition.GetAssemblyName()
   at ICSharpCode.Decompiler.Metadata.DotNetCorePathFinderExtensions.DetectTargetFrameworkId(PEReader assembly, String assemblyPath) in D:\GitWorkspace\ILSpy_72\ICSharpCode.Decompiler\Metadata\DotNetCorePathFinderExtensions.cs:line 56
   at ICSharpCode.Decompiler.Console.ILSpyCmdProgram.DecompileAsProject(String assemblyFileName, String outputDirectory) in D:\GitWorkspace\ILSpy_72\ICSharpCode.Decompiler.Console\IlspyCmdProgram.cs:line 217
   at ICSharpCode.Decompiler.Console.ILSpyCmdProgram.OnExecute(CommandLineApplication app) in D:\GitWorkspace\ILSpy_72\ICSharpCode.Decompiler.Console\IlspyCmdProgram.cs:line 101
```

In the previous version, 7.2.0.6702-preview2, the issue was not present yet.

To reproduce the issue, I used the following commands:
```PowerShell
dotnet tool uninstall ilspycmd -g
dotnet tool install ilspycmd --version <version> -g
rm -Recurse <output-directory>\*
ilspycmd <assembly-name>.resources.dll -p -o <output-directory>
```

A workaround is to use an older version of ilspycmd.

**Comment by siegfriedpammer** [comment]
The error message hints at the following change: https://github.com/icsharpcode/ILSpy/commit/db6b08548494195b75240066151a0dd2be2e45e0

However, this has been present in ilspycmd ever since version 7.1. Would you be able to provide an example for us to try to reproduce this error? Thanks!

**Comment by pvoosten** [comment]
I crafted a simple example. The zip file contains the solution to create the contained satellite assemblies. The satellite assemblies can be found in *Ilspy2691\ClasslibExample\bin\Debug\net6.0\nl* or *Ilspy2691\ClasslibExample\bin\Debug\net6.0\de*

[Ilspy2691.zip](https://github.com/icsharpcode/ILSpy/files/8703139/Ilspy2691.zip)

**Comment by pvoosten** [comment]
The fix for #2417 indeed caused the issue. Another workaround is to change *%userprofile%\.dotnet\tools\.store\ilspycmd\7.2.1.6856\ilspycmd\7.2.1.6856\tools\net6.0\any\ilspycmd.runtimeconfig.json*:

```json
{
  "runtimeOptions": {
    "tfm": "net6.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "6.0.0"
    },
    "configProperties": {
      "System.GC.Server": true,
      "System.Globalization.Invariant": false, // was true
      "System.Globalization.PredefinedCulturesOnly": true,
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
    }
  }
}
```

**Comment by siegfriedpammer** [comment]
Thank you for reporting this.
